console/EAP7-1159_Support_runtime_operations_for_messaging_servers.adoc

```diff
@@ -0,0 +1,100 @@
+= Support runtime operations for messaging servers
+:author:            Harald Pehl
+:email:             hpehl@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+:issue-base-url:    https://issues.jboss.org/browse/
+
+== Overview
+
+There are lots of new runtime operations for messaging servers, which are currently not supported in HAL. This RFE is about to add support for these operations.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/EAP7-1159[EAP7-1159]
+
+=== Dev Contacts
+
+* mailto:hpehl@redhat.com[Harald Pehl]
+
+=== QE Contacts
+
+* mailto:pjelinek@redhat.com[Pavel Jelinek]
+
+=== Affected Projects or Components
+
+* HAL
+
+== Requirements
+
+The console adds the following actions to the messaging server runtime column:
+
+* View: Will open a new application screen (see below)
+* `export-journal()`
+* `import-journal(String xmlFile)`: Does this operation support file uploads?
+* `force-failover()`
+* `reset-all-message-counter-histories()`
+* `reset-all-message-counters()`
+
+The new application screen will show various information about the selected messaging server. The following operations will be supported:
+
+=== Connections
+
+* `list-connection-ids()`
+* `list-connections-as-json()`
```

Comment:
for historical reasons, some management operations have 2 variants, one with a DMR output, the other with JSON (and sometimes HTML).

I don't think it is necessary to expose both to HAL. Exposing the DRM-variant is enough

Reply:
I cannot find an equivalent DMR-variant. `list-connection-ids()` lists only the connection IDs, whereas `list-connections-as-json()` returns more details like `clientAddress` or `sessionCount`.